src/openrct2/Editor.cpp

```diff
@@ -38,32 +39,7 @@
 
 namespace Editor
 {
-    static uint8 _editorSelectedRides[MAX_RIDE_OBJECTS];
-    static uint8 _editorSelectedSmallScenery[MAX_SMALL_SCENERY_OBJECTS];
-    static uint8 _editorSelectedLargeScenery[MAX_LARGE_SCENERY_OBJECTS];
-    static uint8 _editorSelectedWalls[MAX_WALL_SCENERY_OBJECTS];
-    static uint8 _editorSelectedBanners[MAX_BANNER_OBJECTS];
-    static uint8 _editorSelectedFootpaths[MAX_PATH_OBJECTS];
-    static uint8 _editorSelectedFootpathAdditions[MAX_PATH_ADDITION_OBJECTS];
-    static uint8 _editorSelectedSceneryGroups[MAX_SCENERY_GROUP_OBJECTS];
-    static uint8 _editorSelectedParkEntrances[MAX_PARK_ENTRANCE_OBJECTS];
-    static uint8 _editorSelectedWaters[MAX_WATER_OBJECTS];
-    static uint8 _editorSelectedStexs[MAX_SCENARIO_TEXT_OBJECTS];
-
-    uint8 * SelectedObjects[OBJECT_TYPE_COUNT] =
-    {
-        _editorSelectedRides,
-        _editorSelectedSmallScenery,
-        _editorSelectedLargeScenery,
-        _editorSelectedWalls,
-        _editorSelectedBanners,
-        _editorSelectedFootpaths,
-        _editorSelectedFootpathAdditions,
-        _editorSelectedSceneryGroups,
-        _editorSelectedParkEntrances,
-        _editorSelectedWaters,
-        _editorSelectedStexs,
-    };
+    static std::array<std::vector<uint8>, OBJECT_TYPE_COUNT> _editorSelectedObjectFlags;
```

Comment:
Xcode appears to choke on the header for `std::array` not being included (`<array>`).